src/ray/common/status.cc

```diff
@@ -80,7 +81,8 @@ std::string Status::CodeAsString() const {
     return STATUS_CODE_OK;
   }
 
-  static std::map<StatusCode, std::string> code_to_str = {
+  // Ensure this is consistent with `str_to_code` in `StringToCode`.
+  static absl::flat_hash_map<StatusCode, std::string> code_to_str = {
```

Comment:
Why don't we just create the other index using this map on the fly? That's less error prone (than that we should always update 2 places).

Reply:
maybe we can also take out the static variable too and make it const

Reply:
`static const` seems better, but this needs to be `static`. Otherwise it is created and destroyed for each call unnecessarily.

Reply:
You two have different commnets: @mwtian want to put maps in functions(https://github.com/ray-project/ray/pull/20047#discussion_r743032906) while @rkooo567 want to take them out.

@rkooo567 If we use one map to create another, the second one would not be `const`, is it right?

I think these two points are tiny comments about code style or somethin' but it should be better for you two to align with each other.

Reply:
Would a helper function that returns const map work?

Reply:
I think coding style is pretty important if the project wants to last long. And some issues are not just about style, but may affect correctness. @rkooo567 and I should definitely align. My thinking is that a variable should use the smallest scope possible, and there should be minimal runtime overhead here. I personally don't think adding complexity to generate status code string -> code value map is worth it, because passing status code by string instead of by integer value is something we should move away from (inefficient, not a good practice).

Reply:
glad to hear that